app_dart/lib/src/request_handlers/dart_internal_subscription.dart

```diff
@@ -2,31 +2,208 @@
 /// The PubSub subscription is set up here:
 /// https://console.cloud.google.com/cloudpubsub/subscription/detail/dart-internal-build-results-sub?project=flutter-dashboard
 @immutable
 class DartInternalSubscription extends SubscriptionHandler {
-  /// Creates an endpoint for listening to dart-internal build results.
+  /// Creates an endpoint for listening for dart-internal build results.
+  /// The message should contain a single buildbucket id
   const DartInternalSubscription({
     required super.cache,
     required super.config,
     super.authProvider,
+    required this.buildBucketClient,
+    @visibleForTesting this.datastoreProvider = DatastoreService.defaultProvider,
+    this.retryOptions = Config.buildbucketRetry,
   }) : super(subscriptionName: 'dart-internal-build-results-sub');
 
+  final BuildBucketClient buildBucketClient;
+  final DatastoreServiceProvider datastoreProvider;
+  final RetryOptions retryOptions;
+
   @override
   Future<Body> post() async {
-    log.fine('Consumed message: ${message.data}');
+    final DatastoreService datastore = datastoreProvider(config.db);
+
+    log.info("Converting message data to non nullable int");
+
+    final int buildbucketId = int.parse(message.data.toString().replaceAll("'", ""));
+
+    log.info("Creating build request object");
+    final GetBuildRequest request = GetBuildRequest(id: buildbucketId.toString());
+
+    log.info(
+      "Calling buildbucket api to get build data for build $buildbucketId",
+    );
+    late Build build;
+    try {
+      build = await retryOptions.retry(
+        () async {
+          return _getBuildFromBuildbucket(request);
+        },
+        retryIf: (Exception e) => e is UnfinishedBuildException,
+      );
+    } catch (e) {
+      if (e is! UnfinishedBuildException) {
+        log.severe(
+          "Failed to get build data for build $buildbucketId with the exception: ${e.toString()}",
+        );
+        throw InternalServerError(
+          'Failed to get build number $buildbucketId from Buildbucket. Error: $e',
+        );
+      }
+    }
+
+    log.info("Checking for existing task in datastore");
+    late Task? existingTask;
+    existingTask = await _getExistingTaskFromDatastore(build, datastore);
+
+    late Task taskToInsert;
+    if (existingTask != null) {
+      log.info("Creating Task from existing Task");
+      existingTask.buildNumber = build.number;
+      existingTask.buildNumberList = '${build.number},${existingTask.buildNumberList}';
+      taskToInsert = existingTask;
+    } else {
+      log.info("Creating Task from Buildbucket result");
+      taskToInsert = await _createTaskFromBuildbucketResult(build, datastore);
+    }
+
+    log.info("Inserting Task into the datastore: ${taskToInsert.toString()}");
+    await datastore.insert(<Task>[taskToInsert]);
+
+    return Body.forJson(taskToInsert.toString());
+  }
+
+  Future<Build> _getBuildFromBuildbucket(
+    GetBuildRequest request,
+  ) async {
+    late Build build;
```

Comment:
Here and everywhere, do not use late when you are initializing in the next line.